### PR TITLE
[rnnt] add RNN prediction network

### PR DESCRIPTION
**Pull request by zhiyun**

### CODE REVIEW

File: axlearn/audio/asr_decoder.py [hunk 1]
```diff
@@ -476,3 +477,69 @@ def _map_label_sequences(inputs: Tensor, *, blank_id: int = 0, pad_id: int = 0)
...
+        self._add_child("rnn", cfg.rnn_cell.set(input_dim=cfg.emb_dim, output_dim=cfg.output_dim))
+
+    def forward(self, inputs: Tensor) -> Tensor:
+        """Computes prediction network output from the inputs.
+
+        Args:
+            inputs: An int Tensor of shape [batch_size, num_labels]. Valid tokens are in the range
+                [0, vocab_size). Out-of-range token ids are clamped to the bounds of the array.
+            See https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html
+            #out-of-bounds-indexing.
```
Comment (markblee) [inline axlearn/audio/asr_decoder.py:516]:
```suggestion
                See https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html
            #out-of-bounds-indexing.
```

File: axlearn/audio/asr_decoder.py [hunk 2]
```diff
@@ -476,3 +477,69 @@ def _map_label_sequences(inputs: Tensor, *, blank_id: int = 0, pad_id: int = 0)
...
+
+        Returns: A Tensor of shape [batch_size, num_labels, output_dim].
```
Comment (markblee) [inline axlearn/audio/asr_decoder.py:518]:
```suggestion
        Returns:
            A Tensor of shape [batch_size, num_labels, output_dim].
```

File: axlearn/audio/asr_decoder.py (hunk 1 above)
Comment (zhiyun) [reply axlearn/audio/asr_decoder.py]: Done.

File: axlearn/audio/asr_decoder.py (hunk 2 above)
Comment (zhiyun) [reply axlearn/audio/asr_decoder.py]: Done.

**Review by markblee — APPROVED** [review]

**Pull request merged**